### PR TITLE
Clarify Java 17 upcoming breaking change

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
related to #7107 

## Fixed Issue(s)
N/A

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by zilm13** [comment]
Is it a reason jdk16 images should be specially covered?

**Comment by StefanBratanov** [comment]
> Is it a reason jdk16 images should be specially covered?

Don't know, it's just something that was available that won't be anymore. Doubt there are many users using it but would be worth to mention it.

**Comment by zilm13** [comment]
Ahh I got it, if you use something like `develop-jdk16` and pull it regularly to update, it will be no more supported. We could change it to smth like "that `jdk16` images will no longer be available, use `jdk17` instead", I like suggestions more than prohibitions :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Upcoming Breaking Changes
 
-- Upgrading source code to Java 17 meaning users will need to upgrade their Java install to at least 17, or use the jdk17 variant of the docker image.
+- Upgrading source code to Java 17 which means users will need to upgrade their Java installation to at least 17. Docker users will not be affected, apart from the change that `jdk16` images will no longer be available for new Teku versions.
```
Comment (rolfyone) [inline CHANGELOG.md]:
```suggestion
- Upgrading the minimum Java version to Java 17, which means users will need to upgrade their Java installation to at least `Java 17`.  The docker versions relying on `jdk16` will no longer be published, so docker users explicitly referencing the `jdk16` build need to update their package to reference `jdk17`, as tags `develop-jdk16`, `develop-jdk16-arm64`, `latest-jdk16` will no longer be able to be updated. 
```

Comment (StefanBratanov) [reply CHANGELOG.md]: Thanks that's good. Done.

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**